File: Hangman.java
package hangman;
import java.util.Scanner;
public class Hangman {

	private int remValues[] = new int[26];
	public String playerNames[] = {"",""};
	public String player1String = "";
	public String player2String = "";
	public int numberOfAttemptsLeft = 7;
	private char inputChar = '\0';
	private int success = 0;
	
	public void gameStart(Scanner s) {
		System.out.println("\n\t\tHANGMAN WORLD\n\n");
		System.out.print("Enter the player 1 name: ");
		playerNames[0] = s.nextLine().toUpperCase();
		System.out.print("Enter the player 2 name: ");
		playerNames[1] = s.nextLine().toUpperCase();
	}
	
	public void takePlayer1String(Scanner s) {
		System.out.print(playerNames[0]+" Enter the string: ");
		player1String = s.nextLine().toUpperCase();
		for(int i = 0;i<30;i++)
			System.out.println();
		for(int i = 0; i<player1String.length();i++) {
			player2String += "_";
			System.out.print(" "+player2String.charAt(i));
		}
	}
	
	public void takeInputPlayer2(Scanner s) {
		System.out.print(playerNames[1] +" Enter the character: "); 
		String inputString = s.next().toUpperCase();
		inputChar = inputString.charAt(0);
	}
	
	private void inPlay() {
		int matched = 0;
		for(int i = 0; i<player1String.length();i++) {
			if(inputChar == player1String.charAt(i)) {
				matched = 1;
				player2String = player2String.substring(0, i) + inputChar + player2String.substring(i+1, player1String.length());
			}
			System.out.print(" "+player2String.charAt(i));
		}
		remValues[(int)(inputChar) - 65] = 1;
		if(matched==0)
			numberOfAttemptsLeft--;		
		if(player1String.equals(player2String))
			success = 1;		
	}
	
	public void printValues() {
		for(int i = 0; i<26; i++) {
			if(i%5==0)
				System.out.println();
			if(i==25)
				System.out.print("    ");
			if(remValues[i]==0)			
				System.out.print(" "+(char)(65+i));
			else
				System.out.print(" -");
		}
		System.out.println("\nNumber of attempts left: "+numberOfAttemptsLeft);
	}
	
	public void endGame(Scanner s) {
		while(true) {
			this.takeInputPlayer2(s);
			this.inPlay();
			this.printValues();
			if(numberOfAttemptsLeft==0 && success==0) { 
				System.out.println("\t\tGame Over");
				System.out.println(playerNames[0]+ " entered this: "+player1String);
				System.out.println("Better Luck Next Time!!");
				break;
			}
			if(success == 1) {
				System.out.println("!!Well Played "+playerNames[1] +"!!");
				System.out.println("You Guessed it correct!");
				System.out.println(player1String+"\n");
				break;
			}
		}
	}
	
	public void launchGame(Scanner s) {
		this.gameStart(s);
		this.takePlayer1String(s);
		this.printValues();
		this.endGame(s);
	}
	
	public static void main(String[] args) {
		Scanner s = new Scanner(System.in);
		Hangman game1 = new Hangman();
		game1.launchGame(s);
	}
}

File: Hangman.cpp
#include<iostream>
#include<string>

using namespace std;

class Game1 {

	private:
		int remValues[26]={0};
		string playerNames[2] = {"",""};
		string player1String = "";
		string player2String = "";
		int numberOfAttemptsLeft = 7;
		char inputChar = '\0';
		int success = 0;
	
		void gameStart() {
			cout<<"\n\t\tHANGMAN WORLD\n\n"<<endl;
			cout<<"\tEnter the player 1 name: ";
			getline(cin,playerNames[0]);
			int i;
			for(i=0; i<playerNames[0].length();i++)
				playerNames[0][i] = toupper(playerNames[0][i]); 
			cout<<"\tEnter the player 2 name: ";
			getline(cin,playerNames[1]);
			for(i=0; i<playerNames[1].length();i++)
				playerNames[1][i] = toupper(playerNames[1][i]); 
		}	
	
		void takePlayer1String() {
			cout<<"\t"<<playerNames[0]<<" Enter the string: ";
			getline(cin,player1String);
			int i;
			for(i=0; i<player1String.length();i++)
				player1String[i] = toupper(player1String[i]);
			system("cls");
			for(i=0;i<5;i++)
				cout<<endl;
			cout<<"\t";
			player2String = player1String;
			for(i = 0; i<player1String.length();i++) {
				player2String[i]='_';
				cout<<" "<<player2String[i];
			}
		}	
	
		void takeInputPlayer2() {
			cout<<"\t"<<playerNames[1]<<" Enter the character: "; 
			string inputString;
			getline(cin,inputString);
			inputChar = toupper(inputString[0]);
			system("cls");
			for(int i=0;i<5;i++)
				cout<<endl;
		} 
		
	
		void inPlay() {
			int matched = 0;
			int i;
			cout<<"\t";
			for(i = 0; i<player1String.length();i++) {
				if(inputChar == player1String[i]) {
					matched = 1;
					player2String[i] = inputChar;
				}
				cout<<" "<<player2String[i];
			}
			remValues[(int)(inputChar) - 65] = 1;
			if(matched==0)
				numberOfAttemptsLeft--;		
			if(player1String.compare(player2String)==0)
				success = 1;		
		}	
	
		void printValues() {
			cout<<"\t";
			for(int i = 0; i<26; i++) {
				if(i%5==0){
					cout<<endl;
					cout<<"\t";
				}
				if(i==25)
					cout<<("    ");
				if(remValues[i]==0)			
					cout<<" "<<(char)(65+i);
				else
					cout<<" _";
				}
			cout<<"\n\tNumber of attempts left: "<<numberOfAttemptsLeft<<endl;
		}
	
		void endGame() {
		while(true) {
			this->takeInputPlayer2();
			this->inPlay();
			this->printValues();
			if(numberOfAttemptsLeft==0 && success==0) { 
				cout<<"\n\n\t\tGame Over"<<endl;
				cout<<"\t"<<playerNames[0]<<" entered this: "<<player1String<<endl;
				cout<<"\tBetter Luck Next Time!!"<<endl;
				break;
			}
			if(success == 1) {
				cout<<"\n\n\t!!Well Played "<<playerNames[1]<<"!!"<<endl;
				cout<<"\tYou Guessed it correct!"<<endl;
				cout<<"\t"<<player1String<<endl;
				break;
			}
		}
		}
	
	public:
		void launchGame() {
		this->gameStart();
		this->takePlayer1String();
		this->printValues();
		this->endGame();
	}
};

int main() {
	string ans;
	do{
		system("cls");
		Game1 g1;
		g1.launchGame();
		cout<<"\n\tDo you want to play again?(y/n): ";
		getline(cin,ans);
	}while(toupper(ans[0])=='Y');
	system("cls");
	cout<<"\tThanks for Playing!!"<<endl;
	cout<<"\n\tHope You Liked The Game!!"<<endl;
}


